packages/catalog-server/tsconfig.json

```diff
@@ -0,0 +1,30 @@
+{
+  "compilerOptions": {
+    "incremental": true,
+    "composite": true,
+    "tsBuildInfoFile": "./tsconfig.tsbuildinfo",
+    "target": "es2021", 
+    "module": "esnext", 
+    "sourceMap": true,
+    "outDir": "./", 
+    "rootDir": "./src",
+    "strict": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+    "noUncheckedIndexedAccess": true,
+    "noPropertyAccessFromIndexSignature": true,
+    "moduleResolution": "node", 
+    "esModuleInterop": true,
+    "experimentalDecorators": true,
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true 
+  },
+  "include": ["src/**/*.ts"],
+  "exclude": [],
+  "references": [
+    { "path": "../catalog-api" },
+    { "path": "../custom-elements-manifest-tools" }
+  ]
```

Comment:
This precludes the use of wireit. Intentional?

Reply:
I'm trying to preserve jump-to-definition across packages. Builds via Wireit seem to work fine with this.

Reply:
I guess it will be ok as of recent Wireit versions, actually. Here's the situation I was worried about:

- Project A is in state 1
- Build project A with wireit
- Modify project A to state 2
- Build project B with wireit, where B references A, causing project A to be built without wireit knowing
- Modify project A back to state 1
- Build project A with wireit

This is potentially a bad state: wireit thinks the output is already correct, since as far as it knows project A was already built at state 1. But actually it's built for state 2. But, since we now do output verification, this scenario will actually work out fine. Wireit will say "hey! the output changed" and do a fresh build.

Reply:
Err, there could be some weird similar situations, or races that happen where wireit thinks it has a lock on building project A, but actually project B is simultaneously implicitly building it without wireit knowing...

I guess my guess it is this will be fine the vast majority of the time, but there is a small chance of something getting weird sometime.

Reply:
But don't you specifically have to pass `--build` to `tsc` to build referenced projects? w/o that you just get incremental and referenced project support in IDEs.

Reply:
Oh, if that's the case then that's great. Are we sure about it? I haven't tested.

Reply:
https://www.typescriptlang.org/docs/handbook/project-references.html#caveats-for-project-references

> Additionally, to preserve compatibility with existing build workflows, tsc will not automatically build dependencies unless invoked with the --build switch.

Seems ok. I thought the only problem was you measuring `--build` to be considerably faster than just incremental builds, which seems like either a bug or just really unintuitive. I'm personally willing to tradeoff a slightly slower build to have working IDE features, but maybe esbuild means we don't have to make that tradeoff even.

Reply:
SGTM